### PR TITLE
Misc fixes 2023-05-09

### LINKED ISSUE #8: Use joined arm contigs
Update config to use contigs 2RL, 3RL, X.

### PR DESCRIPTION
**Pull request by alimanfoo**
* Upgrade environment
* Use min_cohort_size and max_cohort_size throughout instead of cohort_size
* Fix bokeh plots not displaying
* Make joined arm contigs work (resolves #8)
* Add config to inputs on some more rules to ensure rebuild if config changes

### CODE REVIEW

**Comment by alimanfoo** [comment]
Hi @sanjaynagi, haven't had time to get this to fully build yet with joined arm contigs, but hopefully should work (or be close :).

File: workflow/Snakefile [hunk 1]
```diff
@@ -5,6 +5,7 @@ import yaml
 # open configuration file
 configfile: "workflow/config.yaml"
 configpath = workflow.configfiles[0]
+# TODO How does this work? Where does the `config` variable come from?
```
Comment (sanjaynagi) [inline workflow/Snakefile:8]: It comes from the `workflow/config.yaml` file. 
snakemake reads the `configfile`  as a dict.

**Review by sanjaynagi — APPROVED** [review]
LGTM :balloon: :balloon:

File: workflow/Snakefile (hunk 1 above)
Comment (alimanfoo) [reply workflow/Snakefile:8]: Thanks - so snakemake automatically reads whatever file is at the location given by `configfile` and then makes that available as a dict as the `config` variable?

**Comment by alimanfoo** [comment]
Thanks @sanjaynagi. CI is failing with a "dead kernel" message, not sure what's happening there, could be the CI runner is running out of memory or something. I might merge here but then try to do a proper build on GCP.

**Comment by sanjaynagi** [comment]
Yeah Think in papermill that's usually a memory error, cos of using 2RL I guess

File: workflow/Snakefile (hunk 1 above)
Comment (sanjaynagi) [reply workflow/Snakefile:8]: exactly

**Pull request merged**